common/app/common/RequestLogger.scala

```diff
@@ -26,20 +27,18 @@ case class RequestLoggerFields(request: Option[RequestHeader], response: Option[
       "Accept-Encoding", // TODO remove if seen after 2021/09/03
     )
 
-    val allHeadersFields = request
-      .map {
-        _.headers.toMap.map {
-          case (headerName, headerValues) => (headerName, headerValues.mkString(","))
-        }
-      }
-      .getOrElse(Map.empty[String, String])
+    val allHeadersFields = request.map(_.headers.toMap.mapV(_.mkString(","))).getOrElse(Map.empty)
+
+    val allowListedHeaders = (for {
+      headerName <- allowListedHeaderNames
+      value <- allHeadersFields.get(headerName)
```

Comment:
The subtle detail here is that [`play.api.mvc.Headers`](https://github.com/playframework/playframework/blob/571f171903f972e7684ee7a3bdc8633eef0ac9dc/core/play/src/main/scala/play/api/mvc/Headers.scala#L89-L90) (the type of `request.headers`) understands and supports the case-insensitivity of HTTP header names, and so even returns a `Map` that is actually extending SortedMap using an Ordering that treats keys insensitively. So when we call `.get(headerName)` on such a map, the key is looked up case-insensitively.